tests/components/dyson/test_vacuum.py

```diff
@@ -1,189 +1,145 @@
+    ATTR_FAN_SPEED,
+    ATTR_FAN_SPEED_LIST,
+    ATTR_STATUS,
+    DOMAIN as VACUUM_DOMAIN,
+    SERVICE_RETURN_TO_BASE,
+    SERVICE_SET_FAN_SPEED,
+    SERVICE_START_PAUSE,
+    SERVICE_STOP,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry
+
+from .common import (
+    ENTITY_NAME,
+    NAME,
+    SERIAL,
+    async_setup_dyson,
+    async_update_entity,
+    get_device,
+)
+
+ENTITY_ID = f"vacuum.{ENTITY_NAME}"
+
+
+@callback
+def _get_vacuum_device() -> Dyson360Eye:
+    device = get_device(Dyson360Eye)
+    device.state = MagicMock()
     device.state.state = Dyson360EyeMode.FULL_CLEAN_RUNNING
     device.state.battery_level = 85
     device.state.power_mode = PowerMode.QUIET
     device.state.position = (0, 0)
     return device
 
 
-def _get_vacuum_device_charging():
-    """Return a vacuum device charging."""
-    device = mock.Mock(spec=Dyson360Eye)
-    device.name = "Device_Vacuum"
-    device.state = mock.MagicMock()
-    device.state.state = Dyson360EyeMode.INACTIVE_CHARGING
-    device.state.battery_level = 40
-    device.state.power_mode = PowerMode.QUIET
-    device.state.position = (0, 0)
-    return device
+async def test_state(hass: HomeAssistant) -> None:
+    """Test the state of the vacuum."""
+    device = await async_setup_dyson(hass, _get_vacuum_device)
 
+    er = await entity_registry.async_get_registry(hass)
+    assert er.async_get(ENTITY_ID).unique_id == SERIAL
 
-def _get_vacuum_device_pause():
-    """Return a vacuum device in pause."""
-    device = mock.MagicMock(spec=Dyson360Eye)
-    device.name = "Device_Vacuum"
-    device.state = mock.MagicMock()
-    device.state.state = Dyson360EyeMode.FULL_CLEAN_PAUSED
-    device.state.battery_level = 40
-    device.state.power_mode = PowerMode.QUIET
-    device.state.position = (0, 0)
-    return device
-
-
-def _get_vacuum_device_unknown_state():
-    """Return a vacuum device with unknown state."""
-    device = mock.Mock(spec=Dyson360Eye)
-    device.name = "Device_Vacuum"
-    device.state = mock.MagicMock()
-    device.state.state = "Unknown"
-    return device
+    state = hass.states.get(ENTITY_ID)
+    assert state.name == NAME
+    assert state.state == STATE_ON
+    attributes = state.attributes
+    assert attributes[ATTR_STATUS] == "Cleaning"
+    assert attributes[ATTR_SUPPORTED_FEATURES] == SUPPORT_DYSON
+    assert attributes[ATTR_BATTERY_LEVEL] == 85
+    assert attributes[ATTR_POSITION] == "(0, 0)"
+    assert attributes[ATTR_FAN_SPEED] == "Quiet"
+    assert attributes[ATTR_FAN_SPEED_LIST] == ["Quiet", "Max"]
 
+    device.state.state = Dyson360EyeMode.INACTIVE_CHARGING
+    device.state.power_mode = PowerMode.MAX
+    state = await async_update_entity(hass, ENTITY_ID)
```

Comment:
Patch the library instead to get access to the callback. Use the callback to trigger a state update on the entity.

https://github.com/home-assistant/core/blob/34a9b93d41396d794b42d08c5bad842f8fe6a372/tests/components/dyson/test_climate.py#L235-L240